src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -50,6 +50,26 @@ public abstract partial class Workspace : IDisposable
         private Action<string> _testMessageLogger;
 
         /// <summary>
+        /// Unloading projects takes a serialization lock, but can also cause Shared Project
+        /// IVsHierarchys to notify us that their context hierarchy has changed. In response to
```

Comment:
"...that their context hierarchy has changed _while we are holding the lock_." since that's the CS101 rule we're violating here.